pkg/tasks/prometheus_user_workload.go

```diff
@@ -187,6 +189,28 @@ func (t *PrometheusUserWorkloadTask) create() error {
 		return errors.Wrap(err, "error creating UserWorkload Prometheus Client GRPC TLS secret")
 	}
 
+	secret, err := t.factory.PrometheusUserWorkloadAdditionalAlertManagerConfigsSecret()
+	if err != nil {
+		return errors.Wrap(err, "initializing UserWorkload Prometheus additionalAlertManagerConfigs secret failed")
+	}
+	if secret != nil {
+		klog.V(4).Info("initializing UserWorkload Prometheus additionalAlertManagerConfigs secret")
+		err = t.client.CreateOrUpdateSecret(secret)
+		if err != nil {
+			return errors.Wrap(err, "reconciling UserWorkload Prometheus additionalAlertManagerConfigs secret failed")
+		}
+	} else {
+		err = t.client.DeleteSecret(&v1.Secret{
```

Comment:
If we never set the additional alertmanager configuration, this will always fail since we will delete an inexistant secret.